presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1816,6 +1819,37 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
         }
     }
 
+    private static class LengthLimitedIterator<T>
+            extends UnmodifiableIterator<T>
+    {
+        private final Iterator<T> iterator;
+        private final long limit;
+        private long count;
+
+        LengthLimitedIterator(final Iterator<T> iterator, final long limit)
+        {
+            this.limit = limit;
+            this.iterator = iterator;
```

Comment:
Flip the order of the assignments, so that they are assigned in the same order as the member variables and parameters